conans/client/rest/file_downloader.py

```diff
@@ -89,21 +99,30 @@ def write_chunks(chunks, path):
         try:
             logger.debug("DOWNLOAD: %s" % url)
             total_length = response.headers.get('content-length') or len(response.content)
-            total_length = int(total_length)
+            total_length = range_start + int(total_length)
             description = "Downloading {}".format(os.path.basename(file_path)) if file_path else None
             progress = progress_bar.Progress(total_length, self._output, description)
 
             chunk_size = 1024 if not file_path else 1024 * 100
             encoding = response.headers.get('content-encoding')
             gzip = (encoding == "gzip")
 
+            # TODO: refactor Progress to allow setting an initial progress
+            progress._processed_size = range_start
```

Comment:
We probably need to add some public methods to ``progress`` to provide the necessary interface

Reply:
Are you planning to do this as part of this PR?

Reply:
Small contribution in my branch: https://github.com/memsharded/conan/tree/feature/download-resume-review